githubapp/client_creator.go

```diff
@@ -229,6 +273,33 @@ func applyMiddleware(base *http.Client, middleware []ClientMiddleware) {
 	}
 }
 
+func cache(cacheFunc func() httpcache.Cache) ClientMiddleware {
+	return func(next http.RoundTripper) http.RoundTripper {
+		return &httpcache.Transport{
+			Transport:           next,
+			Cache:               cacheFunc(),
+			MarkCachedResponses: true,
+		}
+	}
+}
+
+func cacheControl(enabled bool) ClientMiddleware {
+	return func(next http.RoundTripper) http.RoundTripper {
+		return roundTripperFunc(func(r *http.Request) (*http.Response, error) {
+			resp, err := next.RoundTrip(r)
+
+			// Force validation to occur when the cache is disabled by setting max-age=0,
+			// as the cache results will always appear as stale
+			if !enabled {
+				cacheControl := resp.Header.Get("Cache-Control")
+				newCacheControl := maxAgeRegex.ReplaceAll([]byte(cacheControl), []byte("max-age=0"))
+				resp.Header.Set("Cache-Control", string(newCacheControl))
```

Comment:
We should only set this if the original response actually contained a `Cache-Control` header.